components/raftstore/src/store/fsm/apply.rs

```diff
@@ -111,12 +111,8 @@ pub struct PendingCmd<C> {
 }
 
 impl<C> PendingCmd<C> {
-    fn new(index: u64, term: u64, cb: C) -> PendingCmd<C> {
-        PendingCmd {
-            index,
-            term,
-            cb: Some(cb),
-        }
+    fn new(index: u64, term: u64, cb: Option<C>) -> PendingCmd<C> {
```

Comment:
Callback itself can be `Callback::None`, no need to make it an option.

Reply:
It may panic in `drop` as `Callback::None` is an `enum`, so I made it an `option`.

``` rust
impl<C> Drop for PendingCmd<C> {
    fn drop(&mut self) {
        if self.cb.is_some() {
            safe_panic!(
                "callback of pending command at [index: {}, term: {}] is leak",
                self.index,
                self.term
            );
        }
    }
}
```

Reply:
That's because it's not cleared just like other commands in `PendingCmd` before being dropped.

Reply:
Oh, I see, should drain in delegate.destroy.

Reply:
@BusyJay Addressed, PTAL again, thanks!

Reply:
> That's because it's not cleared just like other commands in `PendingCmd` before being dropped.

@BusyJay  I found It is still possible to panic, such as calling `stop_node` in test code, it'll call `batch_system.shutdown` without call `delegate.destroy`.